packages/dds/sequence/src/revertibles.ts

```diff
@@ -158,17 +158,25 @@ export function appendChangeIntervalToRevertibles(
 	revertibles: SharedStringRevertible[],
 ) {
 	const startSeg = previousInterval.start.getSegment() as SharedStringSegment;
+	const startType =
+		startSeg.removedSeq !== undefined
+			? ReferenceType.SlideOnRemove
```

Comment:
```suggestion
			? ReferenceType.SlideOnRemove | ReferenceType.RangeBegin
```

Similarly below for end and for interval delete. This is required for `appendSharedStringDeltaToRevertibles` to properly restore the revertible data if the segment is _later_ removed (which might be in the position they slid to).

Reply:
(This is probably one of the causes of the remaining bugs)